pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -236,6 +248,70 @@ func (pc *PackageControllerClient) Enable(ctx context.Context) error {
 	return nil
 }
 
+// UpdateSecrets is used to update the registry-mirror-cred secret used by the packages controller.
+func (pc *PackageControllerClient) UpdateSecrets(ctx context.Context, client client.Client, cluster *anywherev1.Cluster) error {
+	secretName := "registry-mirror-cred"
+	secretKey := types.NamespacedName{
+		Namespace: constants.EksaPackagesName,
+		Name:      secretName,
+	}
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: constants.EksaPackagesName,
+		},
+	}
+	credErr := client.Get(ctx, secretKey, secret)
+	err := fillRegistrySecret(cluster.Name, cluster.Spec.RegistryMirrorConfiguration, secret)
+	if err != nil {
+		return err
+	}
+
+	if apierrors.IsNotFound(credErr) {
+		return client.Create(ctx, secret)
+	} else if credErr == nil {
+		return client.Update(ctx, secret)
+	}
+	return credErr
+}
+
+func fillRegistrySecret(clusterName string, registry *anywherev1.RegistryMirrorConfiguration, secret *corev1.Secret) error {
+	caDataName := clusterName + "_ca.crt"
+	insecureDataName := clusterName + "_insecure"
+	configName := "config.json"
+	if secret.Data == nil {
+		secret.Data = make(map[string][]byte)
+	}
+	secret.Data[caDataName] = []byte(registry.CACertContent)
+	secret.Data[insecureDataName] = []byte(strconv.FormatBool(registry.InsecureSkipVerify))
+
+	dconfig := &dockerConfig{Auths: make(map[string]*dockerAuth)}
+	configData, ok := secret.Data[configName]
+	if ok {
+		err := json.Unmarshal(configData, dconfig)
+		if err != nil {
+			return err
+		}
+	}
+	username, password, err := config.ReadCredentials()
```

Comment:
why are we reading from env vars here? The controller should avoid at all cost passing data through env vars